### PR TITLE
Upgrade dependency versions

### PR DESCRIPTION
**Pull request by sfc-gh-xhuang**
Copies https://github.com/snowflakedb/snowflake-kafka-connector/pull/797 - merge on behalf of @bansallovish

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -397,6 +397,10 @@
                     <groupId>io.swagger</groupId>
                     <artifactId>swagger-core</artifactId>
                 </exclusion>
+                <exclusion>
```
Comment (sfc-gh-tzhang) [inline pom.xml:400]: @sfc-gh-lsembera Do you know what's the implication of excluding this in other dependencies? I assume we will use the version defined in the ingest sdk (or any other dependencies that uses this)? If we don't exclude this, we will just use a random version that appears to be the first one?

Comment (sfc-gh-tzhang) [reply pom.xml:400]: And what's the best practice here? Should we define the version explicitly?

**Review by sfc-gh-tzhang — APPROVED** [review]
LGTM, could we ask @sfc-gh-lsembera to take a quick look to make sure we're following the best practice?

**Review by sfc-gh-asen — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (sfc-gh-lsembera) [reply pom.xml:400]: If we define the dependency directly, there is no need to exclude it from other dependencies. Maven uses the [_nearest definition_](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#transitive-dependencies) rule, which will pick the version declared directly.

**Pull request merged**